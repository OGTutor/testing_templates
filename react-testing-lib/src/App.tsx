// import { useEffect, useState } from 'react';

// function App() {
// 	const [data, setData] = useState({});
// 	const [toggle, setToggle] = useState(false);
// 	const [value, setValue] = useState('');

// 	const onClick = () => setToggle(prev => !prev);

// 	useEffect(() => {
// 		setTimeout(() => {
// 			setData({});
// 		}, 100);
// 	}, []);

// 	return (
// 		<div>
// 			<h1 data-testid='value-elem'>{value}</h1>
// 			{toggle === true && <div data-testid='toggle-elem'>toggle</div>}
// 			{data && <div style={{ color: 'red' }}>data</div>}
// 			<h1>Hello world</h1>
// 			<button data-testid='toggle-btn' onClick={onClick}>
// 				Click
// 			</button>
// 			<input
// 				onChange={e => setValue(e.target.value)}
// 				type='text'
// 				placeholder='input value...'
// 			/>
// 		</div>
// 	);
// }

// export default App;

import { FC } from 'react';
import Navbar from './components/Navbar/Navbar';
import AppRouter from './routes/AppRouter';

const App: FC = () => {
	return (
		<div>
			<Navbar />
			<AppRouter />
		</div>
	);
};

export default App;
