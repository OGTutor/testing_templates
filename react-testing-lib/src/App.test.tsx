import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';

describe('TEST APP', () => {
	test('Heading, button & input', () => {
		render(<App />);
		const helloWorldElement = screen.getByText(/hello world/i);
		const buttonElement = screen.getByRole('button');
		const inputElement = screen.getByPlaceholderText(/input value/i);

		expect(helloWorldElement).toBeInTheDocument();
		expect(buttonElement).toBeInTheDocument();
		expect(inputElement).toBeInTheDocument();

		expect(inputElement).toMatchSnapshot();
	});

	test('Query, find & style', async () => {
		render(<App />);
		// eslint-disable-next-line
		screen.debug();
		const helloWorldElementQuery = screen.queryByText(/hello2/i);
		expect(helloWorldElementQuery).toBeNull();
		const helloWorldElementFind = await screen.findByText(/data/i);
		expect(helloWorldElementFind).toBeInTheDocument();
		expect(helloWorldElementFind).toHaveStyle({ color: 'red' });
		// eslint-disable-next-line
		screen.debug();
	});

	test('INPUT, CLICK EVENT & get', () => {
		render(<App />);
		// input
		// eslint-disable-next-line

		const input = screen.getByPlaceholderText(/input value/i);
		expect(screen.queryByTestId('value-elem')).toContainHTML('');
		fireEvent.input(input, {
			target: { value: 'value' },
		});
		// USER ACTION EMULATION

		// userEvent.type(input, 'value');
		expect(screen.queryByTestId('value-elem')).toContainHTML('value');

		// button
		// eslint-disable-next-line

		const button = screen.getByTestId('toggle-btn');
		expect(screen.queryByTestId('toggle-elem')).toBeNull();
		fireEvent.click(button);
		expect(screen.getByTestId('toggle-elem')).toBeInTheDocument();
		fireEvent.click(button);
		expect(screen.queryByTestId('toggle-elem')).toBeNull();
	});
});
