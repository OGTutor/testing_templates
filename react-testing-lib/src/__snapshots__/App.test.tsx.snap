// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders learn react link 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`TEST APP Heading 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`TEST APP Heading, button & input 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`TEST APP renders learn react link 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`TEST APP renders learn react link 2`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
